Ginger/GingerTest/Variable Dependancies/VariableDependancyTest.cs

```diff
@@ -0,0 +1,229 @@
+            activity3.Active = true;
+
+            ActDummy actDummy5 = new ActDummy();
+            actDummy5.Active = true;
+            ActDummy actDummy6 = new ActDummy();
+            actDummy6.Active = true;
+
+            activity3.Acts.Add(actDummy5);
+            activity3.Acts.Add(actDummy6);
+
+            activityGroup.AddActivityToGroup(activity1);
+            activityGroup.AddActivityToGroup(activity2);
+            activityGroup.AddActivityToGroup(activity3);
+
+            BF1.Activities.Add(activity1);
+            BF1.Activities.Add(activity2);
+            BF1.Activities.Add(activity3);
+
+            BF1.ActivitiesGroups.Add(activityGroup);
+
+            Platform p = new Platform();
+            p.PlatformType = ePlatformType.Web;
+            BF1.TargetApplications.Add(new TargetApplication() { AppName = "SCM" });
+
+            mGR = new GingerRunner();
+            mGR.Name = "Test Runner";
+            mGR.CurrentSolution = new Ginger.SolutionGeneral.Solution();
+
+            ProjEnvironment environment = new ProjEnvironment();
+            environment.Name = "Default";
+            BF1.Environment = environment.Name;
+
+            Agent a = new Agent();
+            a.DriverType = Agent.eDriverType.SeleniumChrome;
+
+            mGR.SolutionAgents = new ObservableList<Agent>();
+            mGR.SolutionAgents.Add(a);
+
+            mGR.ApplicationAgents.Add(new ApplicationAgent() { AppName = "SCM", Agent = a });
+            mGR.SolutionApplications = new ObservableList<ApplicationPlatform>();
+
+            mGR.SolutionApplications.Add(new ApplicationPlatform() { AppName = "SCM", Platform = ePlatformType.Web, Description = "New application" });
+            mGR.BusinessFlows.Add(BF1);
+
+            Context context1 = new Context();
+            context1.BusinessFlow = BF1;
+            context1.Activity = BF1.Activities[0];
+
+            mGR.CurrentBusinessFlow = BF1;
+            mGR.CurrentBusinessFlow.CurrentActivity = BF1.Activities[0];
+            mGR.Context = context1;
+        }
+
+        [TestMethod]
+        //[Timeout(60000)]
+        public void ActivityVariablesDependancyTest()
+        {
+            //Arrange
+            ObservableList<Activity> activityList = BF1.Activities;
+
+            BF1.EnableActivitiesVariablesDependenciesControl = true;
+
+            //Added selection list variable in BF
+            VariableSelectionList selectionList = new VariableSelectionList();
+            selectionList.OptionalValuesList.Add(new OptionalValue("a"));
+            selectionList.OptionalValuesList.Add(new OptionalValue("b"));
+            selectionList.SelectedValue = selectionList.OptionalValuesList[0].Value;
+            selectionList.ItemName = "MyVar";
+            BF1.Variables.Add(selectionList);
+
+            //Added dependancies in activities
+            string[] variableValues, variableValue;
+
+            variableValues = new string[] { "a", "b" };
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4817688)